core/src/main/java/com/linecorp/armeria/client/retry/HttpStatusBasedRetryStrategy.java

```diff
@@ -38,11 +39,20 @@
 final class HttpStatusBasedRetryStrategy implements RetryStrategy<HttpRequest, HttpResponse> {
 
     private final List<HttpStatus> retryStatuses;
+    private final Backoff backoff;
```

Comment:
Could we allow a user specify a different backoff for a different HTTP status? i.e. `Map<HttpStatus, Backoff>`?